src/commands/registries/azure/tasks/buildImageInAzure.ts

```diff
@@ -5,12 +5,29 @@
 
 import * as vscode from 'vscode';
 import { IActionContext, UserCancelledError } from '@microsoft/vscode-azext-utils';
+import type { Run } from '@azure/arm-containerregistry';
 import { scheduleRunRequest } from './scheduleRunRequest';
+import { delay } from '../../../../utils/promiseUtils';
+import { getArmContainerRegistry } from '../../../../utils/lazyPackages';
 
-export async function buildImageInAzure(context: IActionContext, uri?: vscode.Uri | undefined): Promise<void> {
+const WAIT_MS = 5000;
+
+export async function buildImageInAzure(context: IActionContext, uri?: vscode.Uri | undefined): Promise<Run | undefined> {
     if (!vscode.workspace.isTrusted) {
         throw new UserCancelledError('enforceTrust');
     }
 
-    await scheduleRunRequest(context, "DockerBuildRequest", uri);
+    const getRun = await scheduleRunRequest(context, "DockerBuildRequest", uri);
+    
+    let run = await getRun();
+    const { KnownRunStatus } = await getArmContainerRegistry();
+    while (run.status === KnownRunStatus.Running) {
```

Comment:
One more thing I just thought of--there's some additional pre-finish run statuses, namely, `Queued` and `Started`. Should this `while` condition allow those? Otherwise, if the run hasn't gotten to the point of `Running` before reaching this loop, the whole loop will be skipped.

Reply:
Updated it to handle those appropriately!